core/src/main/java/tech/pegasys/eth2signer/core/service/operations/SignForIdentifier.java

```diff
@@ -48,25 +42,26 @@ public SignForIdentifierHandler(
     this.type = type;
   }
 
-  @Override
-  public void handle(RoutingContext routingContext) {
-    final RequestParameters params = routingContext.get("parsedParameters");
-    final String identifier = params.pathParameter("identifier").toString();
+  public SignResponse sign(final String identifier, final String data) {
     final Optional<ArtifactSigner> signer = signerProvider.getSigner(identifier);
     if (signer.isEmpty()) {
       LOG.trace("Unsuitable handler for {}, invoking next handler", identifier);
-      routingContext.next();
-      return;
+      return new SignResponse(SIGNER_NOT_FOUND, identifier);
     }
 
-    final Bytes dataToSign = getDataToSign(params);
+    final Bytes dataToSign;
+    try {
+      if (StringUtils.isBlank(data)) {
```

Comment:
Is this actually invalid? Doesn't seem that useful, but I think it's a valid to sign empty data. Perhaps it doesn't matter if we guard against this though as nobody is realistically going to be signing with no data.

Reply:
Looks like the signature we returning for this case wasn't valid so I think it's fine to have this check in